lib/release/release.go

```diff
@@ -0,0 +1,206 @@
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package release
+
+import (
+	"context"
+	"crypto/tls"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"os"
+
+	"github.com/gravitational/roundtrip"
+	"github.com/gravitational/teleport/api/types"
+	"github.com/gravitational/trace"
+	log "github.com/sirupsen/logrus"
+)
+
+// ClientConfig contains configuration for the release client
+type ClientConfig struct {
+	// TLSConfig is the client TLS configuration
+	TLSConfig *tls.Config
+	// ReleaseServerAddr is the address of the release server
+	ReleaseServerAddr string
+}
+
+// CheckAndSetDefaults checks and sets default config values
+func (c *ClientConfig) CheckAndSetDefaults() error {
+	if c.TLSConfig == nil {
+		return trace.BadParameter("missing TLS configuration")
+	}
+
+	if c.ReleaseServerAddr == "" {
+		return trace.BadParameter("missing release server address")
+	}
+
+	return nil
+}
+
+// Client is a client to make HTTPS requests to the
+// release server using the Teleport Enterprise license
+// as authentication
+type Client struct {
+	// client is the client used to make calls to the release API
+	client *roundtrip.Client
+}
+
+// NewClient returns a new release client with a client
+// to make https requests to the release server
+func NewClient(cfg ClientConfig) (*Client, error) {
+	err := cfg.CheckAndSetDefaults()
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	httpClient := &http.Client{
+		Transport: &http.Transport{
+			TLSClientConfig: cfg.TLSConfig,
+		},
+	}
+
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	client, err := roundtrip.NewClient(fmt.Sprintf("https://%s", cfg.ReleaseServerAddr), "", roundtrip.HTTPClient(httpClient))
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	return &Client{
+		client: client,
+	}, nil
+}
+
+// ListRelease calls the release server and returns a list of releases
+func (c *Client) ListReleases(ctx context.Context) ([]*types.Release, error) {
+	if c.client == nil {
+		return nil, trace.BadParameter("client not initialized")
+	}
+
+	resp, err := c.client.Get(ctx, c.client.Endpoint(types.EnterpriseReleaseEndpoint), nil)
+	if err != nil {
+		log.WithError(err).Error()
+		return nil, trace.Wrap(err)
+	}
+
+	if resp.Code() == http.StatusUnauthorized {
+		return nil, trace.AccessDenied("access denied by the release server")
+	}
+
+	if resp.Code() != http.StatusOK {
+		return nil, trace.Errorf("release server responded with status %d", resp.Code())
+	}
+
+	var releases []Release
+	err = json.Unmarshal(resp.Bytes(), &releases)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	responseReleases := []*types.Release{}
+	for _, r := range releases {
+		releaseAssets := []*types.Asset{}
+		for _, a := range r.Assets {
+			releaseAssets = append(releaseAssets, &types.Asset{
+				Arch:        a.Arch,
+				Description: a.Description,
+				Name:        a.Name,
+				OS:          a.OS,
+				SHA256:      a.SHA256,
+				Size_:       a.Size,
```

Comment:
Can we just rename the Size field in Proto to `AssetSize`  ?
```suggestion
				AssetSize:       a.Size,
```